### ISSUE TITLE
0.8 Compatibility 

### ISSUE THREAD

**Issue by serby** [serby/ctrl#95]
Control needs to be upgraded to node 0.8 but connect currently only works in 0.6.

Discuss

@domharrington @bengourley @aduncan88

**Comment by bengourley** [comment]
Are all dependencies 0.8 compatible?

**Comment by serby** [comment]
No. but connect is the show stopper.

**Comment by bengourley** [comment]
Then the only way to go to 0.8 is to use something else, or wait...

**Comment by microadam** [comment]
yeah sounds like the only logical solution. Seeing as express is heavily coupled with connect and we use express currently, we either replace express or wait. I would have thought replacing express would be a pain

**Comment by evlun** [comment]
Control and its bundles rely on Express/Connect for everything from middleware to session handling to view rendering to `res.redirect` and `next(new Error())`. I don't think it would be feasible to replace Express any time soon.

What is it that actually doesn't work in Connect on 0.8?

**Comment by bengourley** [comment]
Hold on… what makes connect not work on 0.8? I just did…

`nvm use 0.8`
`git clone git://github.com/senchalabs/connect.git`
`cd connect`
`npm install`
`make test`

```
  ............................................................
  .............................................................
  .................................................

  ✔ 170 tests complete (1967ms)
```

**Comment by bengourley** [comment]
Also, re Erik: there is a flatiron component which is compatible with connect style middleware.

**Comment by evlun** [comment]
@bengourley In that case it might be worth looking into flatiron.

**Comment by bengourley** [comment]
Only if we have to. It looks like connect v0.8 compatibility was a red herring.

Also, I just ran control on v0.8 and it works, we just get a few `path.exists is now called fs.exists` messages

**Comment by tomgco** [comment]
v0.8.8 works like a dream, still get the following warning though.

```
path.existsSync is now called `fs.existsSync`.
```

**Comment by serby** [comment]
This is fixed in the 0.1 branch
